include/boost/json/detail/monotonic_resource.hpp

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright (c) 2020 Krystian Stasiowski (sdkrystian@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/vinniefalco/json
+//
+
+#ifndef BOOST_JSON_DETAIL_MONOTONIC_RESOURCE_HPP
+#define BOOST_JSON_DETAIL_MONOTONIC_RESOURCE_HPP
+
+#include <boost/json/detail/config.hpp>
+
+#include <cstddef>
+
+namespace boost {
+namespace json {
+namespace detail {
+namespace dummy {
+
+struct class_type {};
+enum unscoped_enumeration_type { };
+enum class scoped_enumeration_type { };
+
+} // dummy
+
+// [support.types] p5: The type max_�align_�t is a trivial
+// standard-layout type whose alignment requirement
+// is at least as great as that of every scalar type.
+struct max_align_t
+{
+    // arithmetic types
+    char a_;
+    char16_t b_;
+    char32_t c_;
+    bool d_;
+    short int e_;
+    int f_;
+    long int g_;
+    long long int h_;
+    wchar_t i_;
+    float j_;
+    double k_;
+    long double l_;
+    // enumeration types
+    dummy::unscoped_enumeration_type m_;
+    dummy::scoped_enumeration_type n_;
+    // pointer types
+    void* o_;
+    char* p_;
+    dummy::class_type* q_;
+    dummy::unscoped_enumeration_type* r_;
+    dummy::scoped_enumeration_type* s_;
+    void(*t_)();
+    // pointer to member types
+    char dummy::class_type::* u_;
+    void (dummy::class_type::*v_)();
+    // nullptr
+    std::nullptr_t w_;
+};
+
+inline
+constexpr
+std::size_t
+max_align()
+{
+    return alignof(
+#ifdef BOOST_JSON_STANDALONE
+        std::
```

Comment:
jesus H this is ugly...never split an identifier's scope with a macro! Use
```
#ifdef BOOST_JSON_STANDALONE
        std::max_align_t
#else
       max_align_t
#endif
```

Reply:
Hah, was being clever. I'll change it.